### PR TITLE
Add implied preprocessor definitions

### LINKED ISSUE #18: Missing implied preprocessor definitions cause clang errors
If `_AFXDLL` is in your preprocessor definitions, you will get the following clang diagnostic error from some windows header(s):

"error: Please use the /MD switch for _AFXDLL builds"

Setting "Use of MFC in a Shared DLL" will get you `_AFXDLL` automatically, but the code that emits this error is looking for `_DLL`. That doesn't normally come from preprocessor definitions explicitly, it's controlled by the project's runtime library setting.

We should check the runtime library setting and add the appropriate implied defines to the preprocessor definitions. There are probably a lot more of these that might turn out to be necessary, but this is the only diagnostic error I've encountered so far. See [here](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros) for a list of candidates.

~~I will probably have to tack my fix on to #16, because it will modify the same code.~~

There is a merge conflict in one of the test json files, it would go in cleanly after the others, so I'll wait on that. In the meantime, you can see the proposed fix [in this commit on my fork.](https://github.com/dakotahawkins/vs-sourcetrail/commit/a6c052aefc00be74328cb747544169d6a7f1ec56#diff-7a8b5a44c49dc7cce2b81f53ff72fe05)

### PR DESCRIPTION
**Pull request by dakotahawkins**
If `_AFXDLL` is in your preprocessor definitions, you will get the following clang diagnostic error from some windows header(s):

"error: Please use the /MD switch for _AFXDLL builds"

Setting "Use of MFC in a Shared DLL" will get you `_AFXDLL` automatically, but the code that emits this error is looking for `_DLL`. That doesn't normally come from preprocessor definitions explicitly, it's controlled by the project's runtime library setting.

This checks the runtime library setting and adds the appropriate implied defines to the preprocessor definitions. There are probably a lot more of these that might turn out to be necessary, but this is the only diagnostic error I've encountered so far. See [here](https://docs.microsoft.com/en-us/cpp/preprocessor/predefined-macros) for a list of candidates.

Fixes #18

### CODE REVIEW

File: SourcetrailExtensionTests/data/all_in_same_folder/test.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [
   {
     "directory": "<CompilationDatabaseFilePath>",
-    "command": "clang-tool -fms-extensions -fms-compatibility -fms-compatibility-version=19  -isystem \"<Macro $(VC_IncludePath)>\"  -isystem \"<Macro $(WindowsSDK_IncludePath)>\"  -D WIN32  -D _DEBUG  -D _CONSOLE  -D _UNICODE  -D UNICODE -std=c++14  \"<ProjectBaseDirectory>/main.cpp\"",
+    "command": "clang-tool -fms-extensions -fms-compatibility -fms-compatibility-version=19  -isystem \"<Macro $(VC_IncludePath)>\"  -isystem \"<Macro $(WindowsSDK_IncludePath)>\"  -D _DEBUG  -D _MT  -D _DLL  -D WIN32  -D _CONSOLE  -D _UNICODE  -D UNICODE -std=c++14  \"<ProjectBaseDirectory>/main.cpp\"",
```
Comment (dakotahawkins) [inline SourcetrailExtensionTests/data/all_in_same_folder/test.json]: Test data now includes implied definitions.

File: VCProjectEngineWrapper/VCCLCompilerToolWrapper.cs [hunk 2]
```diff
@@ -86,7 +86,24 @@ public string[] GetAdditionalIncludeDirectories()
...
+				break;
+			case runtimeLibraryOption.rtMultiThreadedDLL:
+				additionalDefines += "_MT;_DLL;";
+				break;
+			case runtimeLibraryOption.rtMultiThreadedDebugDLL:
+				additionalDefines += "_DEBUG;_MT;_DLL;";
+				break;
+			}
+
+			return (additionalDefines + _wrappedRules.GetEvaluatedPropertyValue("PreprocessorDefinitions")).Split(';');
```
Comment (dakotahawkins) [inline VCProjectEngineWrapper/VCCLCompilerToolWrapper.cs]: Adds implied definitions for the current `RuntimeLibrary` setting based on [this documentation.](https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library)

**Comment by mlangkabel** [comment]
Well done! We may need to regard all the other implicit defines in the future because the actual source code that gets fed into a clang tool may be using them, but for now this is a great start.

**Comment by dakotahawkins** [comment]
@mlangkabel See my suggestion [here](https://github.com/CoatiSoftware/vs-sourcetrail/issues/18#issuecomment-334158547). That may be difficult to test, though I can at least investigate manually whether setting the driver in the compile command database works with a native visual studio command.

**Pull request merged**